dlib/image_processing/full_object_detection.h

```diff
@@ -140,6 +140,10 @@ namespace dlib
         bool ignore = false;
 
         operator rectangle() const { return rect; }
+        bool operator == (const mmod_rect& rhs) const
```

Comment:
This is a little bit funny.  If you are going to implement operator== then it needs to compare for full equality.  So checking all the fields is needed, not just rect.